testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/security/SimpleSessionCookieIT.java

```diff
@@ -32,17 +54,144 @@ public SimpleSessionCookieIT(final String nexusBundleCoordinates) {
+   * Validate Nexus Cookies during Sign-in and Sign-out
+   *
+   * @param nexusUrl the base Nexus URL to validate against
    */
-  @Test
-  public void sessionCookie() throws Exception {
-    HttpGet get = new HttpGet(getSecureUrl() + "service/local/authentication/login");
-    try (CloseableHttpClient client = clientBuilder().build()) {
-      try (CloseableHttpResponse response = client.execute(get, clientContext())) {
+  private void exerciseCookieSpec(final URL nexusUrl) throws Exception {
+
+    // handle cookies like a browser to aid validation
+    final CookieSpec spec = new BrowserCompatSpecFactory().create(null);
+    final CookieOrigin cookieOrigin = cookieOrigin(nexusUrl);
+    final HttpClientContext clientContext = clientContext();
+    final CookieStore cookieStore = new BasicCookieStore();
+    final CredentialsProvider credProvider = credentialsProvider();
+    SetCookie loginCookie;
+
+    try (CloseableHttpClient client = clientBuilder().setDefaultCookieStore(cookieStore).
+        setDefaultCredentialsProvider(credProvider).build()) {
+
+      // 1. login with credentials and get session cookie
+      // Set-Cookie: JSESSIONID=98a766bc-bc33-4b3c-9d9f-d3bb85b0cf00; Path=/nexus; Secure; HttpOnly
+      HttpGet loginGet = new HttpGet(nexusUrl.toExternalForm() + "service/local/authentication/login");
+      try (CloseableHttpResponse response = client.execute(loginGet, clientContext)) {
+        assertThat(response.getStatusLine().getStatusCode(), is(200));
+        assertThat("login cookie should have been stored in the cookie store", cookieStore.getCookies(),
+            hasSize(1));
+        assertThat("expected session cookie in cookie store", getSessionCookie(cookieStore), notNullValue());
+
+        Header[] sessionCookieHeaders = response.getHeaders("Set-Cookie");
+        Header sessionCookieHeader = sessionCookieHeaders[0];
+        List<Cookie> sessionCookies = spec.parse(sessionCookieHeader, cookieOrigin);
+        loginCookie = (SetCookie) sessionCookies.get(0);
+        String headerText = sessionCookieHeader.toString();
+
+        assertCommonSessionCookieAttributes(nexusUrl, loginCookie, headerText);
+        assertThat("expecting one cookie in same Set-Cookie header", sessionCookies, hasSize(1));
+        assertThat("expecting one Set-Cookie header for login", sessionCookieHeaders, arrayWithSize(1));
+        assertThat("login cookie should NOT look like deleteMe cookie", loginCookie.getValue(), not(containsString(
+            "deleteMe")));
+        assertThat("login cookie should not have an expiry date - the browser deletes the session cookie when " +
+                "replaced by a new one by same name from the server OR when the UA decides",
+            loginCookie.isPersistent(), is(false));
+
+        assertThat("login session cookie with valid session id should always be marked HttpOnly",
+            headerText, containsString("; HttpOnly"));
+      }
+      finally {
+        loginGet.reset();
```

Comment:
agree probably not needed due to no reuse, but I am nervous about connection leaks and by this javadoc:

http://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/client/methods/HttpRequestBase.html#releaseConnection()

perhaps I am being too paranoid.